src/main/java/com/google/api/codegen/config/DiscoInterfaceModel.java

```diff
@@ -15,13 +15,16 @@
 package com.google.api.codegen.config;
 
 import com.google.api.codegen.discogapic.transformer.DiscoGapicNamer;
+import com.google.api.codegen.discovery.Document;
 
 /** Discovery-based InterfaceModel. */
 public class DiscoInterfaceModel implements InterfaceModel {
   private final String interfaceName;
+  private final DiscoApiModel apiModel;
 
-  public DiscoInterfaceModel(String interfaceName) {
+  public DiscoInterfaceModel(String interfaceName, Document document) {
     this.interfaceName = interfaceName;
+    this.apiModel = new DiscoApiModel(document);
```

Comment:
Because then I want to do the same thing for ProtoInterfaceModel, but that results in creating a bunch of new ProtoApiModels in some places where it used to be just passing around raw proto Models.